### ISSUE TITLE
csmTransactionRebuild.py bug

### ISSUE THREAD

**Issue by morrone** [IBM/CAST#672]
I tried using csmTransactionRebuild.py found in the ibm-csm-bds package version 1.5.3-2161.  It ran for a while, but then hit the following error:

```
[INFO] Exception caught: 'utf8' codec can't decode byte 0xef in position 17: invalid continuation byte
Traceback (most recent call last):
  File "/opt/ibm/csm/bigdata/beats/csmTransactionRebuild.py", line 192, in dump_allocations
    file.write('{0}\n'.format(json.dumps(data_wrapped, default=str)))
  File "/usr/lib64/python2.7/json/__init__.py", line 250, in dumps
    sort_keys=sort_keys, **kw).encode(obj)
  File "/usr/lib64/python2.7/json/encoder.py", line 207, in encode
    chunks = self.iterencode(o, _one_shot=True)
  File "/usr/lib64/python2.7/json/encoder.py", line 270, in iterencode
    return _iterencode(o, 0)
UnicodeDecodeError: 'utf8' codec can't decode byte 0xef in position 17: invalid continuation byte
```

**Comment by morrone** [comment]
That does not seem to address the issue.  It looks to me like the exception is triggered under json.dumps(), which means that data_wrapped needs to be sanitized before the call to json.dumps() rather than trying to sanitize the results of the call.

I don't know if you have a better method, but this is what I did that seems to work:

```
def sanitize_string(v):
    return v.decode('utf-8', 'ignore')

def sanitize_dict(d):
  for k, v in d.iteritems():
    if isinstance(v, dict):
      d[k] = sanitize_dict(v)
    elif isinstance(v, str):
      d[k] = sanitize_string(v)
    else:
      d[k] = v
  return d
```

And then just before the json.dumps() line:

```
                data_wrapped = sanitize_dict(data_wrapped)
```

**Comment by mew2057** [comment]
This fix was implemented in PR #698. @morrone  your solution was the most succinct so I added it to the script.
